scripts/eqtl_hail_batch/conditional_analysis.py

```diff
@@ -134,29 +134,34 @@ def get_genotype_df(filtered_mt_path, residual_df, gene_snp_test_df):
     mt = mt.filter_cols(set_to_keep.contains(mt['onek1k_id']))
     # Do this only on SNPs contained within gene_snp_df to save on
     # computational time
-    snps_to_keep = set(gene_snp_test_df.snpid)
+    snps_to_keep = set('chr' + gene_snp_test_df.snp_id)
     sorted_snps = sorted(snps_to_keep)
     sorted_snp_positions = list(map(lambda x: x.split(':')[:2][1], sorted_snps))
     sorted_snp_positions = [int(i) for i in sorted_snp_positions]
     # get first and last positions, with 1 added to last position (to make it inclusive)
-    chromosome = gene_snp_test_df.snpid[0].split(':')[:1][0]
+    chromosome = 'chr' + gene_snp_test_df.snp_id[0].split(':')[:1][0]
     first_and_last_snp = chromosome + ':' + str(sorted_snp_positions[0]) + '-' + str(sorted_snp_positions[-1]+1)
     mt = hl.filter_intervals(mt, [hl.parse_locus_interval(first_and_last_snp, reference_genome='GRCh38')])
     t = mt.entries()
     t = t.annotate(n_alt_alleles=t.GT.n_alt_alleles())
     t = t.key_by(contig=t.locus.contig, position=t.locus.position)
     t = t.select(t.alleles, t.n_alt_alleles, sampleid=t.onek1k_id)
-    t = t.annotate(snpid=hl.str(':').join(list(map(hl.str, [t.contig, t.position, t.alleles[0], t.alleles[1]]))))
+    t = t.annotate(snp_id=hl.str(':').join(list(map(hl.str, [t.contig, t.position, t.alleles[0], t.alleles[1]]))))
     # Further reduce the table by only selecting SNPs needed
     set_to_keep = hl.literal(snps_to_keep)
-    t = t.filter(set_to_keep.contains(t['snpid']))
+    t = t.filter(set_to_keep.contains(t['snp_id']))
     # only keep SNPs where all samples have an alt_allele value
-    snps_to_remove = set(t.filter(hl.is_missing(t.n_alt_alleles)).snpid.collect())
+    snps_to_remove = set(t.filter(hl.is_missing(t.n_alt_alleles)).snp_id.collect())
     if len(snps_to_remove) > 0:
-        t = t.filter(~hl.literal(snps_to_remove).contains(t.snpid))
+        t = t.filter(~hl.literal(snps_to_remove).contains(t.snp_id))
 
     genotype_df = t.to_pandas(flatten=True)
     genotype_df.rename({'onek1k_id': 'sampleid'}, axis=1, inplace=True)
+    # chromosome must be turned into a number solely
```

Comment:
Is this change just for cosmetics when displaying, or because we are storing chromosome as a numeric column to make the database more efficient etc? If we want to treat things as numeric, there may need to be some additional transformation here to handle text chromosomes (X, Y, MT).

Note that using numeric encoding also opens a can of worms, as there aren't consistent maps for these chromosomes to numbers - i.e. Plink uses X=23, Y=24, XY=25, MT=26 vs most other software uses X=23, Y=24, MT=25. So we need to make an explicit decision on these encodings and ensure they are used consistently everywhere.

Reply:
This change is to address the specifications outlined in the [GitHub issues doc](https://github.com/populationgenomics/tob-wgs/issues/97). @daniaki and @danielreti might know best about why the change is preferred, and if it's not, I can change it back (it would actually make certain parts of my script cleaner, since hail needs the 'chr' prefix to designate it's GRCh38). However, in saying that, everything here is a string representation of the number, so non-numeric.

Reply:
This format was to match the output I got from Joseph's group, but it should cause any major issues on our end to change it back. Feel free to change it back particularly if it makes other parts of the script cleaner.